src/Utilities/Compiler/PooledObjects/PooledDictionary.cs

```diff
@@ -37,14 +38,21 @@ private PooledDictionary(ObjectPool<PooledDictionary<K, V>>? pool, IEqualityComp
                 this.Clear();
             }
 
-            _pool?.Free(this);
+            _pool?.Free(this, CancellationToken.None);
             return result;
         }
 
-        public void Free()
+        public void Free(CancellationToken cancellationToken)
         {
```

Comment:
We should probably mention that people should use this overload if the PooledDictionary is allocated in one analyzer action but freed in a different analyzer action.

Reply:
I'll do this as a follow-up. I want to try get this change into master so it does not miss the RC2 snap.